forc/src/utils/helpers.rs

```diff
@@ -277,6 +282,57 @@ fn format_warning(err: &sway_core::CompileWarning) {
     eprintln!("{}", DisplayList::from(snippet))
 }
 
+/// Given a start and an end position and an input, determine how much of a window to show in the
+/// error.
+/// Mutates the start and end indexes to be in line with the new slice length.
+///
+/// The library we use doesn't handle auto-windowing and line numbers, so we must manually
+/// calculate the line numbers and match them up with the input window. It is a bit fiddly.t
+fn construct_window<'a>(
+    start: &mut LineCol,
+    end: LineCol,
+    start_ix: &mut usize,
+    end_ix: &mut usize,
+    input: &'a str,
+) -> &'a str {
+    // how many lines to prepend or append to the highlighted region in the window
+    const NUM_LINES_BUFFER: usize = 2;
+
+    let total_lines_in_input = input.chars().filter(|x| *x == '\n').count();
+    let total_lines_of_highlight = end.line - start.line;
+    debug_assert!(total_lines_in_input > total_lines_of_highlight);
+
+    let mut current_line = 0;
+    let mut lines_to_start_of_snippet = 0;
+    let mut calculated_start_ix = None;
+    let mut calculated_end_ix = None;
+    for (ix, character) in input.chars().enumerate() {
+        if character == '\n' {
+            current_line += 1
```

Comment:
I assume you've eyeballed the output and it's good, but it feels like the loop should `continue` here.  The `ix` on line 315 will be at the `\n`, won't it?  Is that right?

Reply:
Those indices are only really used to feed the window to the library, they aren't controlling the actual underlining. Is there a downside to pointing at a newline that you're referring to? I can't think of any -- it just tells the library where to draw the boundaries when rendering

Reply:
Yeah, nah, all good.  It just felt like 'the start' of whatever you're searching for should be at the start of the line, rather than the preceding newline char.